docs/docs/user-guide/training-data-format.mdx

```diff
@@ -0,0 +1,659 @@
+---
+id: training-data-format
+sidebar_label: Training Data Format
+title: Training Data Format
+description: Description of the YAML format for training data
+---
+
+## Overview
+
+### YAML Format
+
+Rasa Open Source 2.0 uses [YAML](https://yaml.org/spec/1.2/spec.html) as
+a unified and extendable way to manage all training data,
+including NLU data, stories and rules. 
+
+With the YAML format, training data can be split over any number of YAML files,
+and every file can contain any kind of data. 
+The training data parser will read the top level keys in each file to decide
+what kind of data is in a section at training time.
+
+The [domain](../glossary#domain) uses
+the same YAML format as the training data and can also be split across
+multiple files or combined with training data in one file. The domain includes
+the definitions for [responses](../core/domains#responses) and [forms](../core/forms).
+See the [documentation for domains](../core/domains) to learn more.
+
+### High Level Structure
+
+Each file can contain one or more **keys** with corresponding training
+data. One file can contain multiple keys, as long as there is not more
+than one of a certain key in a single file. The available keys are:
+
+- `version`
+- `stories`
+- `nlu`
+- `rules`
+- `e2e_tests`
+
+All YAML training data files should specify the `version` key in order
+to be parsed correctly. Training data files with no `version` specified will be assumed to
+be in the format of the latest version of Rasa Open Source. Training data files with a version greater than is currently
```

Comment:
the version is NOT specifying the rasa open source version, but the training data version

Reply:
I'd rather phrase it as (also using active voice)

"If you don't specify a `version` key in your training data file, Rasa will assume you are using the latest training data format specification."

Reply:
Is there info somewhere about training data versions?

Reply:
Well, so here is how it should work technically:
- we do have different training data specifications, each gets a version. first one is 2.0 - if at some point we need to change the training data schema, e.g. because we want to rename of a training data example, we'd increase the version number to 2.1. this is independent of the rasa open source version though. we should only change the version number if the change is not backwards compatible.
- the user specifies the specification their training data adheres to
- if the installed rasa open source version does not support the specified training data version, it should fail. if the user did not specify any format version, rasa open source should use the latest format it supports

Reply:
hopefully, we'll not need to make a lot of use of that version number